Mock or implement symlink

On really new versions of libstdc++, it now exposes the symlink syscall. This prevents hot/cold linking when using that version. This syscall should either be implemented or mocked to do nothing.

#### Screenshots/Output Dumps/Stack Traces

Here's the compile error for it:

Creating cold package with libc,libm,libpros,okapilib [ERRORS]
/usr/local/lib/pros-toolchain/bin/../lib/gcc/arm-none-eabi/10.2.1/../../../../arm-none-eabi/bin/ld: /usr/local/lib/pros-toolchain/bin/../lib/gcc/arm-none-eabi/10.2.1/../../../../arm-none-eabi/lib/thumb/v7-a+simd/softfp/libstdc++.a(cow-fs_ops.o): in function `std::filesystem::create_directory_symlink(std::filesystem::path const&, std::filesystem::path const&, std::error_code&)':
cow-fs_ops.cc:(.text._ZNSt10filesystem24create_directory_symlinkERKNS_4pathES2_RSt10error_code+0x8): undefined reference to `symlink'

--- a/src/system/dev/file_system_stubs.c
+++ b/src/system/dev/file_system_stubs.c
@@ -56,3 +56,28 @@ int _stat(const char* file, struct stat* st) {
 	errno = ENOSYS;
 	return -1;
 }
+
+int symlink(const char* file, const char* linkpath) {
+	errno = ENOSYS;
+	return -1;
+}
+
+ssize_t readlink(const char* pathname, char* buf, size_t bufsiz) {
+	errno = ENOSYS;
+	return NULL;
+}
+
+int truncate(const char* path, off_t length) {
+	errno = ENOSYS;
+	return -1;
+}
+
+int fchmod(int fd, mode_t mode) {
+	errno = ENOSYS;
+	return -1;
+}
+
+int fchmodat(int dirfd, const char* pathname, mode_t mode, int flags) {
+	errno = ENOSYS;
+	return -1;
+}

--- a/src/system/dev/file_system_stubs.c
+++ b/src/system/dev/file_system_stubs.c
@@ -64,7 +64,8 @@ int symlink(const char* file, const char* linkpath) {
 
 ssize_t readlink(const char* pathname, char* buf, size_t bufsiz) {
 	errno = ENOSYS;
-	return NULL;
+	return -1;
+
 }
 
 int truncate(const char* path, off_t length) {

--- a/src/system/dev/file_system_stubs.c
+++ b/src/system/dev/file_system_stubs.c
@@ -32,6 +32,16 @@ int chmod(const char* pathname, mode_t mode) {
 	return -1;
 }
 
+int fchmod(int fd, mode_t mode) {
+	errno = ENOSYS;
+	return -1;
+}
+
+int fchmodat(int dirfd, const char* pathname, mode_t mode, int flags) {
+	errno = ENOSYS;
+	return -1;
+}
+
 long pathconf(const char* path, int name) {
 	errno = ENOSYS;
 	return -1;
@@ -65,20 +75,9 @@ int symlink(const char* file, const char* linkpath) {
 ssize_t readlink(const char* pathname, char* buf, size_t bufsiz) {
 	errno = ENOSYS;
 	return -1;
-
 }
 
 int truncate(const char* path, off_t length) {
-	errno = ENOSYS;
-	return -1;
-}
-
-int fchmod(int fd, mode_t mode) {
-	errno = ENOSYS;
-	return -1;
-}
-
-int fchmodat(int dirfd, const char* pathname, mode_t mode, int flags) {
 	errno = ENOSYS;
 	return -1;
 }

